packages/phone-number-privacy/combiner/src/bls/bls-cryptography-client.ts

```diff
@@ -12,50 +12,95 @@ function flattenSigsArray(sigs: Uint8Array[]) {
   return Uint8Array.from(sigs.reduce((a, b) => a.concat(Array.from(b)), [] as any))
 }
 export class BLSCryptographyClient {
-  private verifiedSignatures: Uint8Array[] = []
-
-  public async addSignature(
-    serviceResponse: ServicePartialSignature,
-    blindedMessage: string,
-    logger?: Logger
-  ): Promise<void> {
-    logger = logger ?? rootLogger
-    const polynomial = config.thresholdSignature.polynomial
-    const sigBuffer = Buffer.from(serviceResponse.signature, 'base64')
+  private unverifiedSignatures: ServicePartialSignature[] = []
+  private verifiedSignatures: ServicePartialSignature[] = []
+  private get totalSignatures(): number {
+    return this.unverifiedSignatures.length + this.verifiedSignatures.length
+  }
+  private get allSignatures(): Uint8Array {
+    const allSigs = this.verifiedSignatures.concat(this.unverifiedSignatures)
+    const sigBuffers = allSigs.map((response) => Buffer.from(response.signature, 'base64'))
+    return flattenSigsArray(sigBuffers)
+  }
 
-    try {
-      await threshold_bls.partialVerifyBlindSignature(
-        Buffer.from(polynomial, 'hex'),
-        Buffer.from(blindedMessage, 'base64'),
-        sigBuffer
-      )
-      this.verifiedSignatures.push(sigBuffer)
-    } catch (err) {
-      logger.error({ url: serviceResponse.url }, ErrorMessage.VERIFY_PARITAL_SIGNATURE_ERROR)
-      logger.error(err)
-    }
+  public addSignature(serviceResponse: ServicePartialSignature) {
+    this.unverifiedSignatures.push(serviceResponse)
   }
 
   /**
    * Returns true if the number of valid signatures is enough to perform a combination
    */
-  public hasSufficientVerifiedSignatures(): boolean {
+  public hasSufficientSignatures(): boolean {
     const threshold = config.thresholdSignature.threshold
-    return this.verifiedSignatures.length >= threshold
+    return this.totalSignatures >= threshold
   }
 
   /*
    * Computes the BLS signature for the blinded phone number.
+   * Throws an exception if not enough valid signatures
+   * and drops the invalid signature for future requests using this instance
    */
-  public async combinePartialBlindedSignatures(): Promise<string> {
+  public async combinePartialBlindedSignatures(
+    blindedMessage: string,
+    logger?: Logger
+  ): Promise<string> {
+    logger = logger ?? rootLogger
     const threshold = config.thresholdSignature.threshold
-    if (!this.hasSufficientVerifiedSignatures()) {
-      const err = new Error(
-        `${ErrorMessage.NOT_ENOUGH_PARTIAL_SIGNATURES} ${this.verifiedSignatures.length}/${threshold}`
+    if (!this.hasSufficientSignatures()) {
+      const error = new Error(
+        `${ErrorMessage.NOT_ENOUGH_PARTIAL_SIGNATURES} ${this.totalSignatures}/${threshold}`
+      )
+      logger.error(error)
```

Comment:
Should this be a structured log? I.e. pass totalSignatures and threshold as a parameter?